### PR TITLE
[testing-on-gke part 6.10] Add script for automating runs

### PR DESCRIPTION
**Pull request by gargnitingoogle**
### Description

This is on top of #2524.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - NA

### CODE REVIEW

File: perfmetrics/scripts/testing_on_gke/examples/run-automated.sh [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+# This script is used purely for automating the run of
+# the script run-gke-tests.sh to
+# run it periodically as a cron-job.
+#
+# For your case, add/remove/modify the configuration parameters as you need.
+#
+# Assumptions for this script to work:
+# 1. You have appropriate access to project_id defined below.
+# 2. You have have the cluster with cluster_name defined below, or enough
```
[the remark this answers is not in the document]
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/run-automated.sh]: doen

File: perfmetrics/scripts/testing_on_gke/examples/run-automated.sh [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+# Some post-run steps to be taken for output collection.
+if test -n "${workload_config}"; then
+  cp ${workload_config} ./workloads.json
+else
+  cp src/gcsfuse/perfmetrics/scripts/testing_on_gke/examples/workloads.json .
+fi
+git -C src/gcsfuse rev-parse HEAD > gcsfuse_commithash
+git -C src/gcs-fuse-csi-driver rev-parse HEAD > gcs_fuse_csi_driver_commithash
+# Fetch cloud-logs for this run. This has not been tested yet.
+# (gcloud logging read --project=${project_id} 'timestamp>="${start_time}"" AND timestamp<="end_time" AND resource.labels.cluster_name="gargnitin-gketesting-us-west1b" ' --order=ASC --format=csv\(timestamp\,resource.labels.pod_name,resource.labels.container_name,"text_payload"\) > cloud_logs.txt) &
```
[the remark this answers is not in the document]
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/run-automated.sh]: done

File: perfmetrics/scripts/testing_on_gke/examples/run-automated.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/run-automated.sh]: done

File: perfmetrics/scripts/testing_on_gke/examples/run-automated.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/run-automated.sh]: done

File: perfmetrics/scripts/testing_on_gke/examples/run-automated.sh [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+# resources in project_id to create this cluster.
+
+# Print all shell commands.
+set -x
+
+# Fail if any command fails.
+set -e
+
+# Define configuration parameters.
+export project_id=gcs-fuse-test-ml
```
Comment (charith87) [inline perfmetrics/scripts/testing_on_gke/examples/run-automated.sh:35]: This should come from the job. They should not be hardcoded here. I would add this as PII.

Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/run-automated.sh]: done.. removed all PII.

File: perfmetrics/scripts/testing_on_gke/examples/workloads_test.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+          "bucket":"The bucket must have objects with name 'train/', 'valid/', and train/img_{i}_of_{numFilesTrain}.npz for every i where i:0-{numFilesTrain}-1 and each train/img_{i}_of_{numFilesTrain}.npz must be of size {recordLength} bytes. The buckets gke-* are all in us-central1, are owned by GKE team and are in their GCP project(s). For best performance, please ensure that the bucket is in the same google-cloud region and GCP project as that of the GKE cluster used for running this test configuration."
+        },
+        {
+          "dlioWorkload": {
+            "numFilesTrain": 1000,
+            "recordLength": 3145728,
+            "batchSizes": [200]
+          },
+          "gcsfuseMountOptions": "implicit-dirs,metadata-cache:ttl-secs:-1,metadata-cache:type-cache-max-size-mb:-1,metadata-cache:stat-cache-max-size-mb:-1,file-cache:max-size-mb:-1,file-cache:cache-file-for-range-read:true,file-cache:enable-parallel-downloads:true",
+          "bucket":"dlio-unet3d-3mb-100k-us-west1",
```
Comment (charith87) [inline perfmetrics/scripts/testing_on_gke/examples/workloads_test.json:52]: few more here. But i am not sure if its difficult to move them out or easy. If its difficult, then probably we submit them now but we cant expose these. We should probably think of moving them to private repo or internal.

Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/workloads_test.json:52]: yeah, these are hard to hide from the repo. Let me figure these out separately.

File: perfmetrics/scripts/testing_on_gke/examples/run-automated.sh [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+else
+  cp src/gcsfuse/perfmetrics/scripts/testing_on_gke/examples/workloads.json .
+fi
+git -C src/gcsfuse rev-parse HEAD > gcsfuse_commithash
+git -C src/gcs-fuse-csi-driver rev-parse HEAD > gcs_fuse_csi_driver_commithash
+# Fetch cloud-logs for this run. This has not been tested yet.
+# (gcloud logging read --project=${project_id} 'timestamp>="${start_time}"" AND timestamp<="${end_time}" AND resource.labels.cluster_name="${cluster_name}" ' --order=ASC --format=csv\(timestamp\,resource.labels.pod_name,resource.labels.container_name,"text_payload"\) > cloud_logs.txt) &
+
+# Upload outputs to GCS after the run.
+output_bucket=gcsfuse-aiml-test-outputs
```
[the remark this answers is not in the document]
Comment (gargnitingoogle) [reply perfmetrics/scripts/testing_on_gke/examples/run-automated.sh]: removed this.

**Review by charith87 — APPROVED** [review]

**Pull request merged**